protobuf/lib/src/protobuf/coded_buffer_writer.dart

```diff
@@ -383,13 +383,17 @@ class CodedBufferWriter {
         _writeInt64(value);
         break;
       case PbFieldType._MESSAGE_BIT:
-        final mark = _startLengthDelimited();
-        value.writeToCodedBufferWriter(this);
-        _endLengthDelimited(mark);
+        writeMessage(value);
         break;
     }
   }
 
+  void writeMessage(GeneratedMessage message) {
```

Comment:
This needs documentation to make clear that the size is being written. According to: https://developers.google.com/protocol-buffers/docs/techniques#streaming an encoded message does not encode its own size.

I think this should be called writeMessageNoTag to correspond to https://github.com/protocolbuffers/protobuf/blob/master/java/core/src/main/java/com/google/protobuf/CodedOutputStream.java#L1238 that does more or less the same thing.